### PR TITLE
Haiku debug build, rdynamic flag unsupported.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -175,10 +175,10 @@ if(NOT DEFINED SNMALLOC_ONLY_HEADER_LIBRARY)
     # Static TLS model unsupported on Haiku
```
Comment (nwf) [inline CMakeLists.txt:175]: Mind updating this comment while you're at it?

**Comment by davidchisnall** [comment]
Is this just a compiler driver bug? Does -Wl,-rdynamic work? If so, that should work everywhere and we can fix at least the clang driver to forward -rdynamic to the linker quite easily to prevent this being a problem for other projects in the future.

**Comment by devnexen** [comment]
It s really unsupported (even what you propose).

**Comment by davidchisnall** [comment]
That’s surprising. -rdynamic just puts everything in the dynamic symbol table. I don’t see how you can have an ELF target that supports shared libraries and doesn’t support it.

**Comment by devnexen** [comment]
All symbols are exported by default on Haiku, it is `-shared` implicitly ; it behaves differently than other oses in that matter.

**Comment by davidchisnall** [comment]
I see, so `-rdynamic` is the default?  A bit surprising it isn't accepted as an argument then.  Can you add that as a comment in this PR?

**Comment by devnexen** [comment]
I think I updated the comment in CMakeLists.txt earlier

File: CMakeLists.txt [hunk 2]
```diff
@@ -172,13 +172,13 @@ if(NOT DEFINED SNMALLOC_ONLY_HEADER_LIBRARY)
     set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS_RELEASE} /DEBUG")
   else()
     add_compile_options(-fno-exceptions -fno-rtti -g -fomit-frame-pointer)
-    # Static TLS model unsupported on Haiku
+    # Static TLS model and -rdynamic flag (always -shared) are unsupported on Haiku
```
Comment (davidchisnall) [inline CMakeLists.txt:175]:
```suggestion
    # Static TLS model is unsupported on Haiku.  
    # All symbols are always dynamic on Haiku and so -rdynamic is redundant (and unsupported).
```

**Review by davidchisnall — APPROVED** [review]

**Pull request merged**